File: HOT100/HOT100p1/6.矩阵/240.搜索二维矩阵-ii.cpp
/*
 * @lc app=leetcode.cn id=240 lang=cpp
 *
 * [240] 搜索二维矩阵 II
 */
#include <bits/stdc++.h>
#include <ranges>
using namespace std;
// @lc code=start
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size(), n = matrix[0].size();
        int i = 0, j = n - 1;
        if (! m || ! n) return 0;
        while (i < m && j >= 0)
        {
            if (matrix[i][j] == target) return 1;
            else if (matrix[i][j] > target) j --;
            else i ++;
        }
        return 0;
    }
};
// @lc code=end

File: HOT100/HOT100p4/240.搜索二维矩阵-ii.py
#
# @lc app=leetcode.cn id=240 lang=python3
#
# [240] 搜索二维矩阵 II
#
from mytools import *
# @lc code=start
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m, n = len(matrix), len(matrix[0])
        i, j = 0, n - 1
        while i < m and j >= 0:
            if matrix[i][j] == target: return True
            elif matrix[i][j] > target: j -= 1
            else: i += 1
            
# @lc code=end

